compiler/block.py

```diff
@@ -413,18 +411,15 @@ def _assign_target(self, target):
   def _register_global(self, node, name):
     var = self.vars.get(name)
     if var:
-      if var.type == Var.TYPE_PARAM:
-        msg = "name '{}' is parameter and global"
-        raise util.ParseError(node, msg.format(name))
-      if var.type == Var.TYPE_LOCAL:
-        msg = "name '{}' is used prior to global declaration"
+      msg = {Var.TYPE_LOCAL: "name '{}' is used prior to global declaration",
```

Comment:
While this is fewer lines, I find that burying the logic in the get() call actually makes the code's intent less obvious.